OfflineLoginRealmAccess/OfflineLoginRealmAccess/Managers/RealmManager.swift

```diff
@@ -0,0 +1,55 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import RealmSwift
+
+class RealmManager: ObservableObject {
+    @Published var realm: Realm?
+
+    // Set the flexible sync configuration for the realm.
+    @MainActor
+    func openRealmWithUser(_ user: User) async throws {
+        // Setting the configuration with `cancelAsyncOpenOnNonFatalErrors` and an initial subscription.
+        let configuration = user.flexibleSyncConfiguration(cancelAsyncOpenOnNonFatalErrors: true,
+                                                           initialSubscriptions: { subs in
+            // Setting the subscription to all public Books and the book associated to the userId
+            subs.append(QuerySubscription<Book> { $0.isPublic || $0.userId == user.id })
+        })
+
+        // Setting the download configuration to `.once` will wait to sync data only the first time the realm
+        // is opened, otherwise this will open the realm immediately and will sync any data on the background
+        // if connected.
+        do {
+            // In case you want to sync your Device Sync data on every realm open, you can change this
+            // to `.always`, which in case of no internet connection will throw into opening the realm
+            // offline,
```

Comment:
We can probably simplify the wording of this a little bit.
```suggestion
            // If you want to sync Device Sync data on every realm open, change this
            // to `.always`. If there is no internet connection, the `catch` block tries to
            // open the realm without syncing first.
```